opal/mca/btl/vader/btl_vader_xpmem.c

```diff
@@ -22,14 +22,37 @@
 
 int mca_btl_vader_xpmem_init (void)
 {
-    mca_btl_vader_component.my_seg_id = xpmem_make (0, VADER_MAX_ADDRESS, XPMEM_PERMIT_MODE, (void *)0666);
+    mca_btl_vader_component.my_seg_id = xpmem_make (0, XPMEM_MAXADDR_SIZE, XPMEM_PERMIT_MODE, (void *)0666);
     if (-1 == mca_btl_vader_component.my_seg_id) {
         return OPAL_ERR_NOT_AVAILABLE;
     }
 
     mca_btl_vader.super.btl_get = mca_btl_vader_get_xpmem;
     mca_btl_vader.super.btl_put = mca_btl_vader_put_xpmem;
 
+    /* Any attachment that goes past the Linux TASK_SIZE will always fail. To prevent this we need to
+     * determine the value of TASK_SIZE. On x86_64 the value was hard-coded in vader to be
+     * 0x7ffffffffffful but this approach does not work with AARCH64 (and possibly other architectures).
+     * Since there is really no way to directly determine the value we can (in all cases?) look through
+     * the mapping for this process to determine what the largest address is. This should be the top
+     * of the stack. No heap allocations should be larger than this value. Since the largest address
+     * may differ between processes the value must be shared as part of the modex and stored in the
+     * endpoint. */
+    FILE *fh = fopen("/proc/self/maps", "r");
```

Comment:
Just to be totally safe, you should check for failure here.

Reply:
Sure. Will do.

Reply:
Done.